internal/pkg/service/common/etcdop/op/atomic.go

```diff
@@ -228,22 +228,23 @@ func (v *AtomicOp[R]) Do(ctx context.Context, opts ...Option) AtomicResult[R] {
 }
 
 func (v *AtomicOp[R]) DoWithoutRetry(ctx context.Context, opts ...Option) (bool, error) {
-	// Create GET operations
-	var getOps []Op
+	tracker := NewTracker(v.client)
+
+	// Create READ operations
+	readTxn := Txn(tracker)
 	for _, opFactory := range v.readPhase {
 		op, err := opFactory(ctx)
 		if err != nil {
 			return false, err
 		}
 		if op != nil {
-			getOps = append(getOps, op)
+			readTxn.Merge(op)
 		}
```

Comment:
Read phase: `Merge` instead of `Then` bellow, and removed unnecessery aux slice.